File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """

        # loop and hold on to the number till you 
        # know for sure you can release it or findme = target - first > cur
        
        left, right = 0, len(numbers) - 1

        while left < right:

            current_sum = numbers[left] + numbers[right]

            if current_sum == target:
            # Return indices as 1-indexed
                return [left + 1, right + 1]
            elif current_sum < target:
                left += 1
            else:
                right -= 1

    # The problem guarantees one solution, so we won't reach this point.
        return []
            






            
        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
    
        let mut left = 0;
        let mut right = numbers.len() - 1;


        while left < right {
            let current_sum = numbers[left] + numbers[right];
            
            if current_sum == target {
                return vec![(left + 1) as i32, (right + 1) as i32];
            } else if current_sum < target {
                left += 1
            } else {
                right -= 1
            }

        }

        vec![]
    }
}